### DISCUSSION TITLE
How do you perform Group Concurrency

### DISCUSSION THREAD

**Discussion by zkldi** [taskforcesh/bullmq#1041] [Q&A]
Hi, I can't seem to find how to do this in the documentation:

I want to have a hard limit on how many jobs can be active *per group*, regardless of workers. For example:

```
// pretend this groupIdConcurrencyLimit is 1
ADD JOB groupId: foo
// foo starts getting processed by worker
ADD JOB groupId: bar
// bar starts getting processed by worker
ADD JOB groupId: foo
// this waits around until the first foo is finished, as the limit per group is only ever one per job.
ADD JOB groupId: baz
// whereas this can start processing immediately 
```

Maybe I'm missing something. Is this possible? Thank you.

**Comment by manast** [comment]
Seems like for some reason many people needs this feature now, anyway, here we have some discussion too: https://github.com/taskforcesh/bullmq/issues/1040
But long story short: it is not available in BullMQ but it will be available in the Pro version soon.

**Reply by zkldi** [reply]
No worries. Sadly the Pro version is prohibitively expensive for an individual dev, but I really appreciate your work nonetheless (and think you deserve the money :P)
